services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/gdpr/KortistoGdprServiceITest.kt

```diff
@@ -319,12 +335,245 @@ class KortistoGdprServiceITest(
+                .single()
+                .prop(GdprError::message)
+                .prop(LocalizedMessage::fi)
+                .contains(hakemus.applicationIdentifier!!)
+        }
+
+        @Test
+        fun `deletes hankekayttaja when there's another admin in the hanke`() {
+            lateinit var kayttaja: HankekayttajaEntity
+            val hanke =
+                hankeFactory.builder(USERID).withHankealue().saveWithYhteystiedot {
+                    kayttaja =
+                        hankeKayttajaService.getKayttajaByUserId(this.hankeEntity.id, USERID)!!
+                    omistaja { addYhteyshenkilo(it, kayttaja) }
+                }
+            val hakemus =
+                hakemusFactory.builder(USERID, hanke).inHandling().saveWithYhteystiedot {
+                    hakija(kayttooikeustaso = Kayttooikeustaso.KAIKKI_OIKEUDET)
+                }
+
+            gdprService.deleteInfo(USERID)
+
+            assertThat(hankeRepository.findAll()).single().prop(HankeEntity::id).isEqualTo(hanke.id)
+            assertThat(applicationRepository.findAll())
+                .single()
+                .prop(ApplicationEntity::id)
+                .isEqualTo(hakemus.id)
+            assertThat(hankeKayttajaService.getKayttajatByHankeId(hanke.id)).single().all {
+                prop(HankeKayttajaDto::kayttooikeustaso).isEqualTo(Kayttooikeustaso.KAIKKI_OIKEUDET)
+                prop(HankeKayttajaDto::sukunimi).isEqualTo(KAYTTAJA_INPUT_HAKIJA.sukunimi)
+                prop(HankeKayttajaDto::id).isNotEqualTo(kayttaja.id)
+            }
+        }
+
+        @Test
+        fun `deletes hankekayttaja when the kayttaja is not an admin`() {
+            lateinit var kayttaja: HankekayttajaEntity
+            val hanke =
+                hankeFactory.builder("Other user").withHankealue().saveWithYhteystiedot {
+                    kayttaja =
+                        kayttaja(
+                            kayttooikeustaso = Kayttooikeustaso.HAKEMUSASIOINTI,
+                            userId = USERID
+                        )
+                    omistaja { addYhteyshenkilo(it, kayttaja) }
+                }
+            val hakemus =
+                hakemusFactory.builder(USERID, hanke).saveWithYhteystiedot {
+                    hakija { addYhteyshenkilo(it, kayttaja) }
+                }
+
+            gdprService.deleteInfo(USERID)
+
+            assertThat(hankeRepository.findAll()).single().prop(HankeEntity::id).isEqualTo(hanke.id)
+            assertThat(applicationRepository.findAll())
+                .single()
+                .prop(ApplicationEntity::id)
+                .isEqualTo(hakemus.id)
+            assertThat(hankeKayttajaService.getKayttajatByHankeId(hanke.id)).single().all {
+                prop(HankeKayttajaDto::kayttooikeustaso).isEqualTo(Kayttooikeustaso.KAIKKI_OIKEUDET)
+                prop(HankeKayttajaDto::sukunimi).isEqualTo(ProfiiliFactory.DEFAULT_NAMES.lastName)
```

Comment:
Here it is a bit unclear as where this `ProfiiliFactory.DEFAULT_NAMES` comes from and why we are comparing to its `lastName`.

Reply:
Yes, this is a bit unclear. I changed this so that the founder is fetched from DB before the delete. Then it's easy to check that the remaining kayttaja has the same id as the founder.